install_script_opw.sh

```diff
@@ -0,0 +1,644 @@
+fi
+
+if [ -n "$TESTING_YUM_URL" ]; then
+  yum_url=$TESTING_YUM_URL
+else
+  yum_url="yum.${repository_url}"
+fi
+
+# We turn off `repo_gpgcheck` for custom REPO_URL, unless explicitly turned
+# on via DD_RPM_REPO_GPGCHECK.
+# There is more logic for redhat/suse in their specific code branches below
+rpm_repo_gpgcheck=
+if [ -n "$DD_RPM_REPO_GPGCHECK" ]; then
+    rpm_repo_gpgcheck=$DD_RPM_REPO_GPGCHECK
+else
+    if [ -n "$REPO_URL" ] || [ -n "$DD_REPO_URL" ]; then
+        rpm_repo_gpgcheck=0
+    fi
+fi
+
+if [ -n "$TESTING_APT_URL" ]; then
+  apt_url=$TESTING_APT_URL
+else
+  apt_url="apt.${repository_url}"
+fi
+
+# TODO: this uses an Agent specific endpoint, remove before using?
+report_failure_url="https://api.datadoghq.com/agent_stats/report_failure"
+if [ -n "$DD_SITE" ]; then
+    report_failure_url="https://api.${DD_SITE}/agent_stats/report_failure"
+fi
+
+telemetry_url="https://instrumentation-telemetry-intake.datadoghq.com/api/v2/apmtelemetry"
+if [ -n "$DD_SITE" ]; then
+    telemetry_url="https://instrumentation-telemetry-intake.${DD_SITE}/api/v2/apmtelemetry"
+fi
+
+if [ -n "$TESTING_REPORT_URL" ]; then
+  report_failure_url=$TESTING_REPORT_URL
+  telemetry_url=$TESTING_REPORT_URL
+fi
+
+worker_flavor="observability-pipelines-worker"
+nice_flavor="Observability Pipelines Worker"
+
+if [ -z "$etcdir" ]; then
+    etcdir="/etc/observability-pipelines-worker"
+fi
+
+if [ -z "$config_file" ]; then
+    config_file="$etcdir/bootstrap.yaml"
+fi
+
+worker_major_version=1
+if [ -n "$DD_WORKER_MAJOR_VERSION" ]; then
+  worker_major_version=${DD_WORKER_MAJOR_VERSION}
+fi
+
+if [ -n "$DD_WORKER_MINOR_VERSION" ]; then
+  # Examples:
+  #  - 20   = defaults to highest patch version x.20.2
+  #  - 20.0 = sets explicit patch version x.20.0
+  # Note: Specifying an invalid minor version will terminate the script.
+  worker_minor_version=${DD_WORKER_MINOR_VERSION}
+fi
+
+worker_dist_channel=stable
+if [ -n "$DD_WORKER_DIST_CHANNEL" ]; then
+  if [ "$repository_url" == "datadoghq.com" ]; then
+    if [ "$DD_WORKER_DIST_CHANNEL" != "stable" ] && [ "$DD_WORKER_DIST_CHANNEL" != "beta" ]; then
+      ERROR_MESSAGE="DD_WORKER_DIST_CHANNEL must be either 'stable' or 'beta'. Current value: $DD_WORKER_DIST_CHANNEL"
+      ERROR_CODE=$INVALID_PARAMETERS_CODE
+      echo "$ERROR_MESSAGE"
+      report_telemetry
+      exit 1;
+    fi
+  elif [ "$DD_WORKER_DIST_CHANNEL" != "stable" ] && [ "$DD_WORKER_DIST_CHANNEL" != "beta" ] && [ "$DD_WORKER_DIST_CHANNEL" != "nightly" ]; then
+    ERROR_MESSAGE="DD_WORKER_DIST_CHANNEL must be either 'stable', 'beta' or 'nightly' on custom repos. Current value: $DD_WORKER_DIST_CHANNEL"
+    ERROR_CODE=$INVALID_PARAMETERS_CODE
+    echo "$ERROR_MESSAGE"
+    report_telemetry
+    exit 1;
+  fi
+  worker_dist_channel=$DD_WORKER_DIST_CHANNEL
+fi
+
+if [ -n "$TESTING_YUM_VERSION_PATH" ]; then
+  yum_version_path=$TESTING_YUM_VERSION_PATH
+else
+  yum_version_path="${worker_dist_channel}/observability-pipelines-worker-1"
```

Comment:
The `-1` should be `-${worker_major_version}`. Same below for the `apt_repo_version`.

Reply:
Good call, lazy hardcoding here 😉

Reply:
Resolved in [5bda0aa](https://github.com/DataDog/agent-linux-install-script/pull/63/commits/5bda0aa8b2b5d7b4c3262460d7012aff26b1a4c5)